File: lib/app/features/core/data/models/media_type.dart
// SPDX-License-Identifier: ice License 1.0

enum MediaType {
  image,
  video,
  audio,
  unknown;

  factory MediaType.fromMimeType(String mimeType) {
    final imageRegex = RegExp('^image/');
    final videoRegex = RegExp('^video/');
    final audioRegex = RegExp('^audio/');

    if (imageRegex.hasMatch(mimeType)) {
      return MediaType.image;
    } else if (videoRegex.hasMatch(mimeType)) {
      return MediaType.video;
    } else if (audioRegex.hasMatch(mimeType)) {
      return MediaType.audio;
    }
    return MediaType.unknown;
  }

  factory MediaType.fromUrl(String url) {
    if (isImageUrl(url)) {
      return MediaType.image;
    } else if (isVideoUrl(url)) {
      return MediaType.video;
    } else if (isAudioUrl(url)) {
      return MediaType.audio;
    } else {
      return MediaType.unknown;
    }
  }

  static bool isImageUrl(String url) {
    return RegExp(r'https?://\S+\.(?:jpg|jpeg|png|gif|bmp|svg|webp)').hasMatch(url);
  }

  static bool isVideoUrl(String url) {
    return RegExp(r'https?://\S+\.(?:mp4|avi|mov|wmv|flv|mkv|webm)').hasMatch(url);
  }

  static bool isAudioUrl(String url) {
    return RegExp(r'https?://\S+\.(?:mp3|wav|ogg|flac|aac|wma)').hasMatch(url);
  }
}

File: ios/NotificationServiceExtension/Translation/Models/MediaType.swift
// SPDX-License-Identifier: ice License 1.0

import Foundation

enum MediaType: String, Codable {
    case image
    case gif
    case video
    case audio
    case document
    case unknown
    
    static func fromMimeType(_ mimeType: String) -> MediaType {
        if mimeType.hasPrefix("image/") {
            return mimeType.contains("gif") ? .gif : .image
        } else if mimeType.hasPrefix("video/") {
            return .video
        } else if mimeType.hasPrefix("audio/") {
            return .audio
        } else if mimeType.hasPrefix("application/") {
            return .document
        }
        
        return .unknown
    }
    
    static func fromUrl(_ url: String) -> MediaType {
        if isImageUrl(url) {
            return .image
        } else if isVideoUrl(url) {
            return .video
        } else if isAudioUrl(url) {
            return .audio
        } else if isDocumentUrl(url) {
            return .document
        }
        
        return .unknown
    }
    
    static func isImageUrl(_ url: String) -> Bool {
        let pattern = "https?://\\S+\\.(?:jpg|jpeg|png|gif|bmp|svg|webp)"
        return url.range(of: pattern, options: .regularExpression) != nil
    }
    
    static func isVideoUrl(_ url: String) -> Bool {
        let pattern = "https?://\\S+\\.(?:mp4|avi|mov|wmv|flv|mkv|webm)"
        return url.range(of: pattern, options: .regularExpression) != nil
    }
    
    static func isAudioUrl(_ url: String) -> Bool {
        let pattern = "https?://\\S+\\.(?:mp3|wav|ogg|flac|aac|wma)"
        return url.range(of: pattern, options: .regularExpression) != nil
    }
    
    static func isDocumentUrl(_ url: String) -> Bool {
        let pattern = "https?://\\S+\\.(?:pdf|doc|docx|xls|xlsx|ppt|pptx|txt|rtf)"
        return url.range(of: pattern, options: .regularExpression) != nil
    }
}
